### ISSUE TITLE
How to set custom user attributes automatically from enrollment form?

### ISSUE THREAD

**Issue by marekhotshot** [goauthentik/authentik#7879]
**Describe your question/**
We need to set custom attributes "first_name" and "last_name" for all users. This was missing by default so we had to create these attributes. (one of our applications (AHA!) that we want to authenticate with requires full name).

If we create user manually or add these attributes manually to existing user in Authentik, it will authenticate correctly. Problem is only for newly created users that we invite and register through registration form.

**Screenshots**
![image](https://github.com/goauthentik/authentik/assets/119857479/7b25257d-631e-4fa2-8346-cd8f868bd649)
![image](https://github.com/goauthentik/authentik/assets/119857479/88d3e0e2-df48-4c65-b976-2ff8832049a0)
![image](https://github.com/goauthentik/authentik/assets/119857479/481ee220-8334-4a96-8e6d-791dea30c8f1)
![image](https://github.com/goauthentik/authentik/assets/119857479/f38dd0b2-8d0c-4a17-b335-b5d9daba0133)
![image](https://github.com/goauthentik/authentik/assets/119857479/a935db65-c10e-43f4-bfc0-97898fbae381)
![image](https://github.com/goauthentik/authentik/assets/119857479/c48a63fa-7cf1-47d0-b7d2-9d8035c98634)
With this setup user in AHA! was created with first_name: default and last_name: default.

Basically we need to somehow get these properties set from registration form automatically to be in this place:
(on the image bellow it was added manually)
![image](https://github.com/goauthentik/authentik/assets/119857479/fb1b25df-753e-4720-b2aa-dc4f70da15ba)

Can you please advise how to achieve this?

**Version and Deployment:**

-   authentik version: 2023.10.4
-   Deployment: Kubernetes

**Comment by marekhotshot** [comment]
issue was in this part:
![image](https://github.com/goauthentik/authentik/assets/119857479/037e683d-295f-4585-a9f5-3b85416450c5)

correct format was: 

 attributes.first_name
 attributes.last_name
 
then custom attributes were written to user...

**Comment by partofthething** [comment]
This was helpful. To further clarify, you have to set the "Field key" on the Prompt edit form to start with `attributes.`, like  `attributes.first_name`, in your case. 

![Image](https://github.com/user-attachments/assets/2faa890a-59e6-46ef-9098-058bc1cc4fd2)

Then after the write stage, the attributes will show up as expected on the user:

![Image](https://github.com/user-attachments/assets/023a31d8-1c82-4629-9c34-8b043b924fd7)

**Comment by joeyame** [comment]
For those who landed here trying to set user attributes from an invite link, here is how I was able to do that using what the above users shared:

```yaml
{
  "name": "Yo Mama",
  "email": "yourmom@gmail.com",
  "attributes": {
    "custom_user_attribute_1": "5 GB",
    "custom_user_attribute_2": "15",
  }
}
```

This sets a default email and name and only leaves it to the invitee to come up with a username and password. I adjusted the recommended default login flow to remove the second stage where the user is asked their email and name.

**Comment by AxelTerizaki** [comment]
Hello. I landed here as well trying to set attributes to a user on enrollment.
My use case is that my NAS only accepts LDAP for external auth, but the `uid` attribute needs to be set. When I manually set it on a user in authentik, like `uid: myusername`, users can login to the NAS using their authentik username.

So my question doesn't really relate to custom fields, but on how to add the user's username as an `uid` attribute. Is it possible ?  How should I do that? I'm a little new to authentik so not fully familliar with everything yet.

**Comment by AxelTerizaki** [comment]
Replying to myself as I've found a solution to my issue here : 
https://www.tekonline.com.au/debugging-attribute-persistence-in-authentik-flows-docker-edition/#comment-3075

For those who might stumble upon the same problem :)
